operator/pkg/gateway-api/gateway_reconcile.go

```diff
@@ -125,7 +132,13 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	}
 
 	httpRoutes := r.filterHTTPRoutesByGateway(ctx, gw, httpRouteList.Items)
-	tlsRoutes := r.filterTLSRoutesByGateway(ctx, gw, tlsRouteList.Items)
+
+	// Only filter TLSRoutes if the CRD is installed
+	var tlsRoutes []gatewayv1alpha2.TLSRoute
+	if helpers.HasTLSRouteSupport(r.Client.Scheme()) {
+		tlsRoutes = r.filterTLSRoutesByGateway(ctx, gw, tlsRouteList.Items)
+	}
+
```

Comment:
do we need to perform the check here? If TLSRoute is not installed, the list will be empty, so filtered result will be empty as well right?